src/Microsoft.OpenApi/Models/OpenApiComponents.cs

```diff
@@ -280,5 +281,45 @@ public void SerializeAsV2(IOpenApiWriter writer)
         {
             // Components object does not exist in V2.
         }
+
+        public override IOpenApiReferenceable ResolveReference(JsonPointer pointer, JsonPointer pointerFromRoot = null)
```

Comment:
This shouldn't be supported.  The components object is not allowed to be the target of a reference.  If someone wants a document that contains just components then they can create an external OpenApiDocument and just add the info element as well as a components element.